modules/react/breadcrumbs/lib/BreadcrumbsCurrentItem.tsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import {CSSObject} from '@emotion/styled';
+import {
+  styled,
+  createElemPropsHook,
+  useLocalRef,
+  composeHooks,
+  createSubcomponent,
+} from '@workday/canvas-kit-react/common';
+import {type} from '@workday/canvas-kit-react/tokens';
+import {TooltipContainer} from '@workday/canvas-kit-react/tooltip';
+import {Popper} from '@workday/canvas-kit-react/popup';
+import {
+  useListItemRegister,
+  useOverflowListItemMeasure,
+} from '@workday/canvas-kit-react/collection';
+import {useTruncateTooltip} from './hooks/useTruncateTooltip';
+import {useBreadcrumbsModel} from './hooks/useBreadcrumbsModel';
+
+// default max-width for truncating text
+const DEFAULT_MAX_WIDTH = 350;
+
+export const truncateStyles = (maxWidth: number = DEFAULT_MAX_WIDTH): CSSObject => ({
+  display: 'inline-block',
+  maxWidth,
+  overflow: 'hidden',
+  textOverflow: 'ellipsis',
+  whiteSpace: 'nowrap',
+  minWidth: '94px',
+});
+
+export interface BreadcrumbsCurrentItemProps extends React.HTMLAttributes<HTMLLIElement> {
+  /**
+   * The max-width of the text
+   *
+   * @default 350px
+   */
+  maxWidth?: number;
+}
+
+const ListItem = styled('li')(
+  {
+    display: 'flex',
+    alignItems: 'center',
+    flexDirection: 'column',
+    textAlign: 'left',
+    ...type.levels.subtext.large,
+    fontWeight: 500,
+  },
+
+  ({maxWidth}: BreadcrumbsCurrentItemProps) => ({
+    ...truncateStyles(maxWidth),
+  })
+);
+
+export const useBreadcrumbsItem = composeHooks(
+  createElemPropsHook(useBreadcrumbsModel)((_model: any, ref: any) => {
+    const {localRef} = useLocalRef(ref);
+    return {ref: localRef};
+  }),
+  useOverflowListItemMeasure,
+  useListItemRegister
+);
+
+export const BreadcrumbsCurrentItem = createSubcomponent('li')({
+  displayName: 'Breadcrumbs.Item',
+  modelHook: useBreadcrumbsModel,
+  elemPropsHook: useBreadcrumbsItem,
+})<BreadcrumbsCurrentItemProps & {ref?: React.RefObject<HTMLLIElement>}>(
+  ({children, maxWidth, ref, ...elemProps}, Element) => {
+    const {
+      isTooltipOpen,
+      openTooltip,
+      closeTooltip,
+      shouldShowTooltip,
+      tooltipProps,
+    } = useTruncateTooltip(ref);
```

Comment:
I'm open to feedback here but wanted to get some thoughts/ ask some questions! I had a similar situation and I do like the approach of a hook for scenarios like this. I did something similar where I wrapped by label text in a `OverflowTooltip` in a `Pill` component. Here's the [reference](https://github.com/Workday/canvas-kit/blob/master/modules/preview-react/pill/lib/PillLabel.tsx#L24). This way we expose the tooltip props as well in case a consumer wants access to it. I'm not saying one approach is better than the other but maybe we should try and define a pattern because this can be useful! What are your thoughts?

Reply:
I remember Nicholas running into this, this is in Tabs: https://github.com/Workday/canvas-kit/blob/master/modules/react/tabs/lib/TabsItem.tsx#L207

Reply:
I agree. This hooks have been removed! Thanks.